### PR TITLE
[small][agg_v2] Fix identifier mapping - layout caching, and MVHashMap storing

### PR DESCRIPTION
**Pull request by igor-aptos**

### CODE REVIEW

File: third_party/move/move-vm/runtime/src/loader/mod.rs [hunk 1]
```diff
@@ -1993,68 +2007,62 @@ impl Loader {
...
+                )
             },
             Type::Struct { name, .. } => {
                 *count += 1;
-                MoveTypeLayout::Struct(self.struct_name_to_type_layout(
-                    name,
-                    &[],
-                    count,
-                    has_identifier_mappings,
-                    depth,
```
Comment (igor-aptos) [inline third_party/move/move-vm/runtime/src/loader/mod.rs:2046]: is it correct to not use depth+1 here? is this only vectorr depth, but not object depth? should it be renamed?

Comment (igor-aptos) [reply third_party/move/move-vm/runtime/src/loader/mod.rs:2046]: @runtian-zhou

**Review by gelash — APPROVED** [review]
LGTM, but indeed @runtian-zhou please have a look?

File: third_party/move/move-vm/runtime/src/loader/mod.rs (hunk 1 above)
Comment (georgemitenkov) [reply third_party/move/move-vm/runtime/src/loader/mod.rs:2046]: But the depth is incremented when processing each field in a struct?

**Review by georgemitenkov — APPROVED** [review]

File: third_party/move/move-vm/runtime/src/loader/mod.rs [hunk 2]
```diff
@@ -1719,25 +1719,30 @@ impl Script {
     }
 }
 
+#[derive(Clone)]
+struct StructLayoutInfoCacheItem {
+    struct_layout: MoveStructLayout,
+    node_count: u64,
+    has_identifier_mappings: bool,
```
Comment (vusirikala) [inline third_party/move/move-vm/runtime/src/loader/mod.rs:1726]: What are node_count and has_identifier_mappings?

Comment (igor-aptos) [reply third_party/move/move-vm/runtime/src/loader/mod.rs:1716]: number of recursive nodes, and whether struct has a field with (aggregator) identifier

**Pull request merged**